Editor/ObjectTagsEditor.cs

```diff
@@ -4,40 +4,132 @@
             ITagsApiManager tagsApiManager = TagsApiManagerEditorFinder.FindTagApiManager();
-            selectTagsSection = await SelectTagsSection.Create(tagsApiManager, objectTags.tags);
+            selectTagsSection = await SelectTagsSection.Create(tagsApiManager, GetCommonTags());
+            selectTagsSection.onTagDeletion = OnTagDeletion;
+            selectTagsSection.onTagAddition = OnTagAddition;
+            selectTagsSection.onTagReset = OnTagReset;
         }
 
         public override void OnInspectorGUI()
         {
+            serializedObject.Update();
             EditorGUILayout.PropertyField(serializedObject.FindProperty("m_Script"), true, new GUILayoutOption[0]);
+            if (!AreAllTagsListsEqual())
+            {
+                ShowWarning();
+                ShowDifferentTags();
+            }
             ShowCurrentTags();
             EditorGUILayout.Space();
             selectTagsSection?.ShowTagsToAdd();
-            if (GUI.changed)
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        private void OnTagDeletion(Tag deletedTag)
+        {
+            objectTagsList.ForEach(objectTags =>
+            {
+                bool isTagAlreadyPresent = objectTags.tags.Any(t => t.id == deletedTag.id);
+                if (isTagAlreadyPresent)
+                {
+                    objectTags.RemoveTag(deletedTag);
+                    EditorUtility.SetDirty(objectTags);
+                }
+            });
+        }
+
+        private void OnTagAddition(Tag addedTag)
+        {
+            objectTagsList.ForEach(objectTags =>
             {
+                bool isTagAlreadyPresent = objectTags.tags.Any(t => t.id == addedTag.id);
+                if (!isTagAlreadyPresent)
+                {
+                    objectTags.AddTag(addedTag);
+                    EditorUtility.SetDirty(objectTags);
+                }
+            });
+        }
+
+        private void OnTagReset()
+        {
+            objectTagsList.ForEach(objectTags =>
+            {
+                objectTags.tags = new List<Tag>();
                 EditorUtility.SetDirty(objectTags);
+            });
+        }
+
+        private List<Tag> GetCommonTags()
+        {
+            HashSet<string> commonTagNames = new HashSet<string>(objectTagsList[0].tags.Select(t => t.name));
+            for (int i = 1; i < objectTagsList.Count; i++)
+            {
+                commonTagNames.IntersectWith(objectTagsList[i].tags.Select(t => t.name));
             }
+
+            return objectTagsList[0].tags
+                .Where(t => commonTagNames.Contains(t.name))
+                .ToList();
+        }
+
+        private List<Tag> GetDifferentTags()
+        {
+            var commonTagNames = new HashSet<string>(GetCommonTags().Select(t => t.name));
+            var differentTags = objectTagsList
+                .SelectMany(objectTags => objectTags.tags)
+                .Where(t => !commonTagNames.Contains(t.name))
```

Comment:
this should work fine, but I would prefer to stick and work with `id`s, it feels safer 
```suggestion
            var commonTagIds = new HashSet<string>(GetCommonTags().Select(t => t.id));
            var differentTags = objectTagsList
                .SelectMany(objectTags => objectTags.tags)
                .Where(t => !commonTagIds.Contains(t.id))
```